### ISSUE TITLE
Test suite fails with Z3 4.6

### ISSUE THREAD

**Issue by romac** [epfl-lara/inox#57]
The inox test suite fails when run against Z3 4.6, as it seems like Z3 now requires explicitly enabling *unsat assumption construction*.

For instance:

```
[info] - 158: simple theorem solvr=smt-z3 lucky=true check=true assum=true assck=false model=1 *** FAILED *** (39 milliseconds)
[info]   inox.package$FatalError: Unexpected error from z3 solver: line 22 column 22: unsat assumptions construction is not enabled, use command (set-option :produce-unsat-assumptions true)
```

**Comment by samarion** [comment]
Closing as it's a duplicate of https://github.com/epfl-lara/inox/issues/32.
The fix depends on having a new published version of scala-smtlib which is pending on @regb.
